src/app/project/compliance/inspection-detail/inspection-detail.component.ts

```diff
@@ -68,58 +82,101 @@ export class InspectionDetailComponent implements OnInit, OnDestroy {
         if (this.compliance.label) {
           this.compliance.label = this.compliance.label.replace(new RegExp('\n', 'g'), '<br />');
         }
-        this.elements = this.compliance.elements;
-        this.tableParams.totalListItems = this.elements.length;
+        // Adding itemClicked to the elements to track the icons
+        this.elements = this.compliance.elements.map(el => {
+          let newEl = {...el, itemClicked: false};
+          return newEl;
+        });
+        this.loading = false;
+      });
+  }
+
+  openElement(element) {
+    this.loading = true;
+    this.handleElementClicked(element);
+    this.searchService.getItem(element._id, 'InspectionElement')
+      .subscribe((res: any) => {
+        if (!res || !res.data) {
+          this.loading = false;
+          return;
+        }
+        this.compliance = new Compliance(res.data);
+        this.submission = res.data;
+        this.submission.description = this.submission.description.replace(new RegExp('\n', 'g'), '<br />');
+
+        this.assets = this.submission.items;
+        for (let i = 0; i < this.assets.length; i++) {
+          this.assets[i].timestamp = new Date(this.assets[i].timestamp);
+        }
+        this.tableParams.totalListItems = this.assets.length;
         this.tableParams.currentPage = 1;
         this.tableParams.pageSize = 100000;
         this.setRowData();
         this.loading = false;
-        this._changeDetectionRef.detectChanges();
+        // this._changeDetectionRef.detectChanges();
+
+        let self = this;
+
+        self.assets.map(async z => {
+          if (z.type === 'photo') {
+            // Show thumb
+            let resource = await self.api.downloadElementThumbnail(self.compliance._id, self.submission._id, z._id);
+            const reader = new FileReader();
+            reader.readAsDataURL(resource);
+            reader.onloadend = function () {
+              // result includes identifier 'data:image/png;base64,' plus the base64 data
+              z.src = reader.result;
+              // self._changeDetectionRef.detectChanges();
+            };
+          } else if (z.type === 'video') {
+            // Show it's type with a clickable event.
+          } else if (z.type === 'voice') {
+            // Show it's type with a clickable event.
+          } else if (z.type === 'text') {
+            // Show it's type with a clickable event.
+          }
+        });
       });
   }
-
+  handleElementClicked(element) {
+    // If an item is clicked need to make sure every other item is unclicked.
+    const newElements = [];
+    this.elements.forEach(i => {
+      if (i._id !== element._id) {
+        i.itemClicked = false;
+      }
+      newElements.push(i);
+    });
+    this.elements = [...newElements];
```

Comment:
No need to assign back. Objects are passed by reference, so the `forEach` is changing the `this.elements` array values directly.

Reply:
Awesome I didn't know that! Thanks for the heads up